### PR TITLE
LPS-153254 Fix focus returns on date input when navigating backwards …

### PR DESCRIPTION
**Pull request by georgel-pop-lr**
…on keyboard

Hi @bryceosterhaus, 

Could you please review this one, as it is an extension of https://github.com/liferay-frontend/liferay-portal/pull/2236

The only issue I have is that destroy is never being called https://github.com/liferay-frontend/liferay-portal/blob/c13a84ab3a1923ca5849b299d328f33fd74aee61/portal-web/docroot/html/taglib/ui/input_date/page.jsp#L274 

As of https://alloyui.com/api/classes/A.DatePicker.html#events destroy should work.

Not sure why is this happening, I would have licked to remove those listeners when not needed.

Regards.

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/custom_fields.js [hunk 1]
```diff
@@ -1270,6 +1270,30 @@ AUI.add(
...
+							delete keysPressed[domEvent.keyCode];
+						}
+					};
+
+					var closePopoverOnKeyboardNavigation = function (instance) {
+						instance.hide();
+
+						keysPressed = {};
+
+						var trigger = A.one('#<%= nameId %>');
```
Comment (bryceosterhaus) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/custom_fields.js:1290]: This won't work, `<%= %>` only works in jsp

**Review by bryceosterhaus — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/custom_fields.js (hunk 1 above)
Comment (georgel-pop-lr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/custom_fields.js:1290]: Ooops thanks, copy/paste going wrong, fixing now.

**Comment by georgel-pop-lr** [comment]
ci:test:sf

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/custom_fields.js (hunk 1 above)
Comment (kresimir-coko) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/custom_fields.js]: I still see an example of this in the changed files @georgel-pop-lr

Comment (georgel-pop-lr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/custom_fields.js]: mmm I did a force push, should be fixed now, could you point out where it is?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/custom_fields.js [hunk 2]
```diff
@@ -1270,14 +1270,46 @@ AUI.add(
...
+
 					var trigger = instance.get('templateNode').one('input');
 
+					var closePopoverOnKeyboardNavigation = function (instance) {
+						instance.hide();
+
+						keysPressed = {};
+
+						if (trigger) {
+							Liferay.Util.focusFormField(trigger);
```
Comment (georgel-pop-lr) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/custom_fields.js:1293]: @kresimir-coko I have changed the file

**Review by bryceosterhaus — APPROVED** [review]
LGTM!

**Comment by georgel-pop-lr** [comment]
ci:forward

**Pull request closed without merging**